lib/pure/parsesql.nim

```diff
@@ -1139,19 +1185,69 @@ proc parseSQL*(input: Stream, filename: string): SqlNode =
   finally:
     close(p)
 
-proc ra(n: SqlNode, s: var string, indent: int)
+proc parseSQL*(input: string, filename=""): SqlNode =
+  ## parses the SQL from `input` into an AST and returns the AST.
+  ## `filename` is only used for error messages.
+  ## Syntax errors raise an `EInvalidSql` exception.
+  parseSQL(newStringStream(input), "")
 
-proc rs(n: SqlNode, s: var string, indent: int,
-        prefix = "(", suffix = ")",
-        sep = ", ") =
+
+type
+  SqlWriter = object
+    indent: int
+    upperCase: bool
+    buffer: string
+
+proc add(s: var SqlWriter, thing: string) =
+  s.buffer.add(thing)
+
+proc add(s: var SqlWriter, thing: char) =
+  s.buffer.add(thing)
+
+proc addKeyw(s: var SqlWriter, thing: string) =
+  if s.buffer.len > 0 and s.buffer[^1] notin " ,\L(":
```

Comment:
```nim

s.buffer[^1] notin {' ', ',', '\L', '('}
```

Reply:
But I'm not a fan of this approach, it's very messy. Decent AST to string handling doesn't require such hacks.

Reply:
What approach would you rather use?